src/field.c

```diff
@@ -551,8 +549,13 @@ void end_field(fieldstruct *field)
     if (field->set)
     {
         for (i=0; i<field->nset; i++)
-            if (field->set[i])
+            if (field->set[i]) {
+                if (field->set[i]->imatab) {
+                    free_tab(field->set[i]->imatab);
+                    field->set[i]->imatab = NULL;
+                }
```

Comment:
Correct me if I am wrong but in principle this part is not needed as this is already taken care of in `end_set()`:
https://github.com/astromatic/scamp/blob/4cf5ff5120cb23618d64c51e7688d1c7a4022d43/src/samples.c#L850-L851

Reply:
Sorry for the long delay!!

You are right, this was not necessary.